pyEpiabm/pyEpiabm/sweep/travel_sweep.py

```diff
@@ -0,0 +1,239 @@
+                density_microcells_list = list(
+                    microcells_to_choose_dict.values())
+                if min(density_microcells_list) < len(
+                        possible_microcell.persons):
+                    # remove old object
+                    microcells_to_choose_dict.pop(list(
+                        microcells_to_choose_dict.keys())[
+                        list(microcells_to_choose_dict.values()).index(
+                            min(density_microcells_list))])
+                    # add new microcell
+                    microcells_to_choose_dict[possible_microcell] = \
+                        len(possible_microcell.persons)
+
+        # Assign to microcell and household in existing population
+        for person in self.initial_cell.persons:
+            selected_microcell = random.choice(list(
+                microcells_to_choose_dict.keys()))
+            # Assign person to microcell and microcell to person
+            selected_microcell.add_person(person)
+            person.microcell = selected_microcell
+            r = random.random()
+            if r < self.travel_params['prob_existing_household']:
+                # Assign to existing household
+                selected_household = random.choice(
+                    selected_microcell.households)
+                selected_household.add_person(person)
+            else:
+                # Create new household
+                selected_microcell.add_household([person])
+
+    def check_leaving_individuals(self, time, person):
+        """
+        Check if individuals travel_end_time is reached
+
+        Parameters
+        ----------
+        time : float
+            Simulation time
+        Person : Person
+            Instance of Person class
+        """
+        remove_individual = False
+        if (hasattr(person, 'travel_end_time')) and \
+                (time > person.travel_end_time):
+            if all([hasattr(person, 'isolation_start_time'),
+                    hasattr(person, 'quarantine_start_time')]):
+                if (person.isolation_start_time and
+                        person.quarantine_start_time is None) or \
+                        (person.isolation_start_time and
+                         person.quarantine_start_time <= time):
+                    remove_individual = True
+            elif hasattr(person, 'isolation_start_time'):
+                if (person.isolation_start_time is None) or \
+                        (person.isolation_start_time <= time):
+                    remove_individual = True
+            elif hasattr(person, 'quarantine_start_time'):
+                if (person.quarantine_start_time is None) or \
+                        (person.quarantine_start_time <= time):
+                    remove_individual = True
+            else:
+                remove_individual = True
+        return remove_individual
+
+    def remove_leaving_individuals(self, time):
```

Comment:
I wasn't sure how to check in one go if travel_end_time exist and is reached and than also check if isolation/quarantine is applied and already over. 
Storing all travelling individuals might be a good idea. Although it could be a big list and I'm not sure if that is preferred over looping over whole population
?

Reply:
I stored them in a `self.travellers` list now and loop over this one to check who should be removed. Removing is now a function of the Person.

Reply:
See below. Created a `self.travellers` list, loop over this list, and created a function in Person to remove the person from cell, microcell and household.